pkg/utils/jsonutils.go

```diff
@@ -58,3 +61,44 @@ func FlattenSingleValue(key string, m map[string]interface{}, child interface{})
 		m[key] = child
 	}
 }
+
+// GetStringFromJson extracts a string value from JSON data based on the provided keys.
+// It uses the supplied workBuf to avoid unnecessary allocations and resizing.
+// If the value is not a string or if an error occurs, it returns an appropriate error.
+func GetStringFromJson(data []byte, workBuf []byte, keys ...string) ([]byte, error) {
+	// Estimate the size needed for workBuf before copying the value
+	estimatedSize := 0
+	for _, key := range keys {
+		if len(key) > estimatedSize {
+			estimatedSize = len(key)
+		}
+	}
+
+	// Resize workBuf to fit the estimated size
+	workBuf = ResizeSlice(workBuf, estimatedSize)
+
+	// Extract the value directly into workBuf
+	var err error
+	var value []byte
+	var dataType jsonparser.ValueType
+
+	value, dataType, _, err = jsonparser.Get(data, keys...)
+	if err != nil {
+		return nil, err
+	}
+
+	if dataType != jsonparser.String {
+		if dataType == jsonparser.Null {
+			return nil, fmt.Errorf("key %s has a null value", strings.Join(keys, ", "))
+		}
+		return nil, fmt.Errorf("expected string value for key(s) %s but got %s", strings.Join(keys, ", "), dataType)
+	}
+
+	// Resize workBuf again to fit the actual length of the extracted value
+	workBuf = ResizeSlice(workBuf, len(value))
+
+	// Copy the extracted value directly into workBuf
+	copy(workBuf, value)
+
+	return workBuf, nil
```

Comment:
You're still only using `workBuf` after you've done all the real work, which defeats the purpose of `workBuf`. I'm pretty sure you'd need to pass it here
```
value, dataType, _, err = jsonparser.Get(data, keys...)
```
for it to be useful (i.e., to prevent allocations). And for that, I think you would have to write an equivalent of `Get()` in our codebase that allows passing a work buffer. Or let me know if I'm missing something